src/transformers/models/mixtral/convert_mixtral_weights_to_hf.py

```diff
@@ -157,16 +157,16 @@ def permute(w, n_heads=n_heads, dim1=dim, dim2=dim):
             state_dict[expert_key + ".weight"] = expert_block.clone()
 
         experts_w2 = [
-            w2[ffn_dim * expert_idx : ffn_dim * (expert_idx + 1), :].contiguous().clone()
+            w2[ffn_dim * expert_idx : ffn_dim * (expert_idx + 1), :].clone(memory_format=torch.contiguous_format)
```

Comment:
is this more efficient than what was there before? i.e what is th emotivation here? 🤗

Reply:
contiguous() may incur a clone if the format is not contiguous, then there are two clones in the old code. There is exactly one clone in the new code.

Reply:
Yes, I think this is more efficient! Maybe `torch.compile` can merge things correctly, but in eager mode calls like `.contiguous().clone()` or `.float().to(device)` will materialize multiple intermediate tensors. When we can do everything in a single call to `clone()` or `to()` we cut down intermediates, which saves time and memory especially when grad is enabled.